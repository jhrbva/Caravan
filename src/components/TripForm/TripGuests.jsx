import React from 'react';
import { Field, Form, Formik } from 'formik';

import '../../utilities/fonts.scss';
import Input from '../Input/Input';
import BigButton from '../BigButton/BigButton';
import {
	required,
} from '../../utilities/formValidation';


const TripGuests = () => (
	<Form>
		<Field
			icon='group_add'
			type='text'
			name='guests'
			placeholder='Guest'
			validate={required}
			autofocus='true'
			component={Input}
		/>
		<BigButton value={'Back'} />
		<BigButton value={'Next'} />
	</Form>
)

export default TripGuests;
