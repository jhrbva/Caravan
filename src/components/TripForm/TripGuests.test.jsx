import React from 'react';
import { shallow } from 'enzyme';
import TripGuests from './TripForm';

describe('TripGuests', () => {
    it('should match the snapshot', () => {
        const wrapper = shallow(<TripGuests />);
        expect(wrapper).toMatchSnapshot();
    });
    it('should have one field', () => {
        const wrapper = shallow(<TripGuests />);
        const fieldWrapper = wrapper.find('Field');
        expect(fieldWrapper.length).toBe(1);
    });
    it('should have an guests field', () => {
        const wrapper = shallow(<TripGuests />);
        const fieldWrapper = wrapper.find('Field');
        const usernameField = fieldWrapper.find({ name: 'guests' });
        expect(usernameField.length).toBe(1);
    });
});
