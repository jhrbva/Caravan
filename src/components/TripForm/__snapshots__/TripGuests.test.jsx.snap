// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripGuests should match the snapshot 1`] = `
<Fragment>
  <Field
    autoFocus={true}
    component={[Function]}
    icon="group_add"
    name="guests"
    placeholder="Guest"
    type="text"
    validate={[Function]}
  />
</Fragment>
`;
